client/core/core.go

```diff
@@ -1065,6 +1075,62 @@ func (c *Core) handleUnbookOrderMsg(dc *dexConnection, msg *msgjson.Message) err
 	return ob.Unbook(&note)
 }
 
+// handleEpochOrderMsg is called when an epoch_order notification is
+// received.
+func (c *Core) handleEpochOrderMsg(dc *dexConnection, msg *msgjson.Message) error {
+	var note msgjson.EpochOrderNote
+	err := json.Unmarshal(msg.Payload, &note)
+	if err != nil {
+		return fmt.Errorf("epoch order note unmarshal error: %v", err)
+	}
+
+	dc.epochsMtx.Lock()
+	defer dc.epochsMtx.Unlock()
+
+	ep, ok := dc.epochs[note.MarketID]
+	if !ok {
+		return fmt.Errorf("no epoch queue found with market id %q",
+			note.MarketID)
+	}
+
+	return ep.Enqueue(&note)
+}
+
+// handleMatchProofMsg is called when a match_proof notification is received.
+func (c *Core) handleMatchProofMsg(dc *dexConnection, msg *msgjson.Message) error {
+	var note msgjson.MatchProofNote
+	err := json.Unmarshal(msg.Payload, &note)
+	if err != nil {
+		return fmt.Errorf("match proof note unmarshal error: %v", err)
+	}
+
```

Comment:
Related to the comment at https://github.com/decred/dcrdex/pull/188#discussion_r384615694, but everything below here could be a method on the `OrderBook`.